### PR TITLE
Invariants: Investigate & fix outstanding critical & potential issues

### PR DESCRIPTION
**Pull request by wischli**
Fixed all outstanding critical and potential issues from #468, June 2025 (pre final v3 audits). Based against intermediary https://github.com/centrifuge/protocol-v3/pull/598 to reduce `main`-rebase noise. Note: There might be new issues emerging from running the suite due to the heavy amount of changes and rebases.

### Invariant Test Improvements

Zero Price Edge Case Handling
- Applied comprehensive zero price handling across all properties
- Prevents division by zero errors in price calculations

Async Vault Properties
- Corrected `maxDeposit` and `maxWithdraw` property calculations
- Fixed `maxMint` property for async vaults
- Added proper escrow edge case handling with meaningful catch cases

Infrastructure Improvements
- Removed mock `MessageProcessor` and `MessageDispatcher` in favor of real implementations
- Fixed asset ID resolution using proper `spoke.vaultToAssetId()` instead of hardcoded **pool currency**
- Updated setup permissions and managers configuration

Code Cleanup
- Removed outdated v2 invariant "fuzzing" suite
- Removed unnecessary `root_{cancel, schedule}Rely` function
- Refactored HubTargets after notify refactor
- Fix broken formatting
- Fix warnings 

### TODO

- [x] Receive review/guidance from Recon
- [x] Run Echidna locally for limited period
- [x] Inspect coverage
- [x] Run with suite & debug potentially emerging issues
- [ ] Format & apply import ordering

### CODE REVIEW

File: test/integration/recon-end-to-end/targets/SpokeTargets.sol [hunk 1]
```diff
@@ -148,20 +148,20 @@ abstract contract SpokeTargets is BaseTargetFunctions, Properties {
 
     // Step 5 - set the request manager
     function spoke_setRequestManager(address vault) public asAdmin {
-        IBaseVault vaultInstance = IBaseVault(_getVault());
+        IBaseVault vaultInstance = IBaseVault(vault);
```
Comment (nican0r) [inline test/integration/recon-end-to-end/targets/SpokeTargets.sol:151]: @wischli could you explain the reason for this change here? 

The initial setup had been created so that the `ReconVaultManager` was the source of truth for the currently set vault. It seems like allowing a fuzzed address here for the vault would make it less likely to correctly set a valid request manager.
